generator/src/main/java/com/scottlogic/deg/generator/generation/AutomatonUtils.java

```diff
@@ -11,54 +11,118 @@
 
     private static final char printableChar = ' ';
 
-    // Returns a string of printable characters based on the supplied automaton.
-    // This method ignores transitions that lead back to the same node so recursive states will only ever produce a
-    // single character. This means for infinite automatons the resulting string isn't the longest possible (as the
-    // longest example would have an infinite length) but is based on the longest path from the start state to
-    // the "furthest" end state.
+    /**
+     * Get the longest string possible given the regex (in the form of the Automaton)
+     * There may be many optional sections within a regex which need to be inspected to calculate the longest possible
+     * string. The automaton has done the hard work of turning the regex into a set of transitions and states.
+     * see https://github.com/ScottLogic/datahelix/blob/master/generator/docs/StringGeneration.md for more detail.
+     *
+     * This method is a recursive implementation that will test each starting point calculating the string and returning
+     * the longest possible variant.
+     *
+     * TODO:
+     * 1) turn this into a non-recursive algorithm
+     * 2) improve performance and/or memory management (don't build strings as part of the process)
+     *
+     * @param automaton The automaton that represents the regex
+     * @return The longest possible string for the given regex/automaton
+     */
     static String getLongestExample(Automaton automaton) {
```

Comment:
I'm not sure about this algorithm. Am I right in thinking it's >O(N), where N is number of either nodes or edges?

Reply:
(should be possible to do longest-path in linear time if we avoid cycles)

Reply:
This has not been implemented to be the most efficient or to be the fastest. There will be much better implementations. 

This implementation does solve the current problem as shown in the additional tests